internal/service/service_test.go

```diff
@@ -0,0 +1,137 @@
+package service_test
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/andreipimenov/golang-training-2021/internal/mock"
+	"github.com/andreipimenov/golang-training-2021/internal/model"
+	"github.com/andreipimenov/golang-training-2021/internal/service"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	validTicker    = "AAPL"
+	validApiKey    = "123"
+	validDate      = "2021-07-26"
+	invalidTicker  = "AAAAAAAAAPL"
+	invalidApiKey  = ""
+	futureDate     = "2022-01-01"
+	weekendDate    = "2021-07-25"
+	stockAPIFormat = "https://www.alphavantage.co/query?function=TIME_SERIES_DAILY&outputsize=full&symbol=%s&apikey=%s"
+)
+
+var (
+	validPrice = model.Price{
+		Open:  "148.2700",
+		High:  "149.8300",
+		Low:   "147.7000",
+		Close: "148.9900",
+	}
+)
+
+func key(ticker string, date time.Time) string {
+	return fmt.Sprintf("%s_%s", ticker, date.Format("2006-01-02"))
+}
+
+type serviceTestSuite struct {
+	suite.Suite
+	s              *service.Service
+	reposytoryMock *mock.Repository
+	clientMock     *mock.HTTPClient
+}
+
+func (suite *serviceTestSuite) SetupTest() {
+	reposytory := &mock.Repository{}
+	client := &mock.HTTPClient{}
+	s := service.New(&zerolog.Logger{}, reposytory, validApiKey)
+	suite.s = s
+	suite.reposytoryMock = reposytory
+	suite.clientMock = client
+}
+
+func TestService(t *testing.T) {
+	suite.Run(t, new(serviceTestSuite))
+}
+
+func (suite *serviceTestSuite) TestHitCache() {
+	d, err := time.Parse("2006-01-02", validDate)
+	suite.NoError(err)
```

Comment:
No need to repeat these 2 lines in every test. It is better if `validDate` is a more convenient value, e.g. `time.Date(2021, 7, 26, 0, 0, 0, 0, time.UTC)`.